### DISCUSSION TITLE
Corrupt (missing) menu item [OER2Go / RACHEL module "en-TED"]

### DISCUSSION THREAD

**Discussion by bernardroth** [iiab/iiab#3124] [General]
I had an issue installing a large package and the menu now has an item I can't remove. It just shows "Attempting to load en-TED" in admin and "en-TED - file not found or improperly formatted" on the main web page. I've tried removing it an reinstalling it and it stays there.

How do I clean it out?

**Comment by tim-moody** [comment]
How did a menu item named en-TED get into the menu? Did you try to add it manually or was it magically added during the package install.

There are some en-ted menu definitions, but nothing en-TED.

You were installing a 'large package'. What is the name of the package; was it a zim?

**Comment by bernardroth** [comment]
I installed it from "Install OER2Go Modules" TED Talks. It showed up after the install.

**Comment by holta** [comment]
(Just as an aside, the page http://oer2go.org leads to http://oer2go.org/viewmod/en-ted_ed_2021 which does not load, so it's hard to get immediate details on this particular RACHEL module!)

**Comment by tim-moody** [comment]
TED Talks: A collection of nearly 2000 short videos on a wide variety of thought-provoking TED  ?

**Reply by bernardroth** [reply]
yes

**Comment by tim-moody** [comment]
OK That sheds some light. There are no existing menu definitions for TED on anything but kiwix.

>It showed up after the install.

can you paste in the contents of /library/www/html/home/menu.json

**Reply by bernardroth** [reply]
{
  "debug": false,
  "menu_lang": "en",
  "force_mobile_layout": false,
  "mobile_header_font": "Open Sans",
  "mobile_incl_description": false,
  "mobile_incl_extra_description": false,
  "mobile_incl_extra_html": false,
  "mobile_incl_footnote": false,
  "desktop_header_font": "Finger Paint",
  "desktop_incl_description": true,
  "desktop_incl_extra_description": true,
  "desktop_incl_extra_html": true,
  "desktop_incl_footnote": true,
  "allow_kiwix_search": true,
  "allow_poweroff": true,
  "poweroff_prompt": "Power Off",
  "autoupdate_menu": true,
  "allow_server_time_update": true,
  "multi_tab": false,
  "tab_names": [
    "Main"
  ],
  "menu_items_1": [
    "en-osm_viewer_v2",
    "en-wikipedia_en_all_maxi",
    "en-wikipedia_en_100",
    "en-wikipedia_en_top_maxi",
    "en-gutenberg_en_all",
    "en-ebooks",
    "en-radiolab",
    "en-ted_en_technology",
    "en-ted_en_science",
    "en-ted_en_playlist-curator-s-picks-top-10-ted-talks-of-2019",
    "en-ted_en_playlist-curator-s-picks-top-10-ted-talks-of-2018",
    "en-ted_en_playlist-curator-s-picks-top-10-ted-talks-of-2017",
    "en-ted_en_top-35-from-2017-2019",
    "en-healthphone",
    "en-hesperian_health",
    "en-kalite",
    "en-wikispecies_en_all_maxi",
    "en-infonet",
    "en-practical_action",
    "en-wiktionary_en_simple_all",
    "en-fitness_stackexchange_com_en_all",
    "en-health_stackexchange_com_en_all",
    "en-nih_rarediseases_en_all_maxi",
    "en-statpearls_en_all_maxi",
    "en-tedmed_en_all",
    "en-teded_en_all",
    "en-wikipedia_en_medicine",
    "en-mediawiki",
    "en-nodered",
    "en-nextcloud",
    "en-wordpress",
    "en-kolibri",
    "en-moodle",
    "en-sugarizer",
    "en-internetarchive",
    "en-calibreweb",
    "en-cdc",
    "en-medline_plus",
    "en-stanford_em",
    "en-asst_medical",
    "en-rpi_guide",
    "en-raspberrypi_stackexchange_com_en_all",
    "en-productivity_stackexchange_com_en_all",
    "en-startups_stackexchange_com_en_all",
    "en-programmers_stackexchange_com_en_all",
    "en-math_stackexchange_com_en_all",
    "en-serverfault_com_en_all",
    "en-scratch",
    "en-wikem",
    "en-lokole",
    "en-sustainability_stackexchange_com_en_all",
    "en-wikinews_en_all_maxi",
    "en-unix_stackexchange_com_en_all",
    "en-avp_stackexchange_com_en_all",
    "en-wikiquote_en_all_maxi",
    "en-wikivoyage_en_all_maxi",
    "en-superuser_com_en_all",
    "en-wikiversity_en_all_maxi",
    "en-wikibooks_en_all_maxi",
    "en-wikisource_en_all_maxi",
    "en-mechanics_stackexchange_com_en_all",
    "en-arduino_stackexchange_com_en_all",
    "en-bioinformatics_stackexchange_com_en_all",
    "en-martialarts_stackexchange_com_en_all",
    "en-quantumcomputing_stackexchange_com_en_all",
    "en-ham_stackexchange_com_en_all",
    "en-3dprinting_stackexchange_com_en_all",
    "en-robotics_stackexchange_com_en_all",
    "en-crypto_stackexchange_com_en_all",
    "en-photo_stackexchange_com_en_all",
    "en-ai_stackexchange_com_en_all",
    "en-movies_stackexchange_com_en_all",
    "en-ted_en_playlist-the-funniest-ted-talks",
    "en-ted_en_playlist-ocean-wonders",
    "en-ted_en_playlist-unsolved-mysteries",
    "en-ted_en_playlist-most-popular-ted-talks-of-2017",
    "en-patents_stackexchange_com_en_all",
    "en-writers_stackexchange_com_en_all",
    "en-gaming_stackexchange_com_en_all",
    "en-crashcourse_en_all",
    "en-ted_en_playlist-9-trippy-ted-talks",
    "en-ted_en_playlist-apocalypse-survival-guide",
    "en-ted_en_playlist-10-talks-from-authors",
    "en-ted_en_playlist-don-t-you-wish-you-lived-here",
    "en-wiktionary_en_all_maxi",
    "en-phet_en",
    "en-PhET",
    "en-openstax",
    "en-saylor",
    "en-computergraphics_stackexchange_com_en_all",
    "en-scicomp_stackexchange_com_en_all",
    "en-cs_stackexchange_com_en_all",
    "en-vi_stackexchange_com_en_all",
    "en-engineering_stackexchange_com_en_all",
    "en-ted_en_playlist-are-we-alone-in-the-universe",
    "en-ted_en_playlist-how-space-technology-can-change-life-on-earth",
    "en-ted_en_playlist-ben-affleck-8-talks-that-amaz",
    "en-ted_en_playlist-bjork-6-talks-that-are-music",
    "en-ted_en_playlist-bono-8-talks-that-give-me-hop",
    "en-ted_en_playlist-how-scientists-learn",
    "en-boundless",
    "en-basic_elecricity_en_all_min",
    "en-phzh_core-english-one_en",
    "en-khan-academy-videos_en_journey-into-cryptography-computer-science-khan-academy",
    "en-keylearning_en_resume-building-with-google-docs",
    "en-khan-academy-videos_en_khan-academy-related-talks-and-interviews",
    "en-Science_in_the_bath_assorted-nonsense",
    "en-Science_in_the_bath_adventures-wonders",
    "en-artofproblemsolving_en_all_maxi",
    "en-appropedia_en_all_maxi",
    "en-Science_in_the_bath_science-in-the-bath",
    "en-mdwiki_en_all",
    "en-computer_videos",
    "en-ted_en_playlist-what-s-happening-to-the-ocean",
    "en-ted_en_playlist-what-s-the-big-deal-about-mars",
    "en-ted_en_playlist-what-happens-when-the-robots-take-our-jobs",
    "en-urban-prepper_en_all",
    "en-khan-academy-videos_en_600-bce-600-ce-second-wave-civilizations-ap-world-history-khan-academy",
    "en-TED",
    "en-ted_en_playlist-what-can-robots-teach-us-about",
    "en-ted_en_playlist-insights-on-terrorism",
    "en-space_stackexchange_com_en_all",
    "en-edison",
    "en-rachelcourses",
    "en-khan-academy-videos_en_lego-robotics-electrical-engineering-khan-academy",
    "en-retrocomputing_stackexchange_com_en_all",
    "en-khan-academy-videos_en_meet-the-professional-computer-programming-khan-academy",
    "en-reverseengineering_stackexchange_com_en_all",
    "en-khan-academy-videos_en_special-relativity-physics-khan-academy",
    "en-khan-academy-videos_en_miscellaneous-health-medicine-khan-academy",
    "en-dsp_stackexchange_com_en_all",
    "en-travel_stackexchange_com_en_all",
    "en-scifi_stackexchange_com_en_all",
    "en-physics_stackexchange_com_en_all",
    "en-beer_stackexchange_com_en_all",
    "en-hardwarerecs_stackexchange_com_en_all",
    "en-khan-academy-videos_en_computer-science",
    "en-khan-academy-videos_en_housing-finance-and-capital-markets-khan-academy",
    "en-money_stackexchange_com_en_all",
    "en-khan-academy-videos_en_foundations-of-american-democracy-us-government-and-civics-khan-academy",
    "en-bicycles_stackexchange_com_en_all",
    "en-khan-academy-videos_en_stocks-and-bonds-finance-and-capital-markets-khan-academy",
    "en-ted_en_playlist-staying-in-geek-out",
    "en-wikem_en_all_maxi",
    "en-wikipedia_en_all_nopic",
    "en-wikipedia_en_for-schools",
    "en-wikisummaries_en_all_maxi",
    "en-ted_en_entertainment",
    "en-ted_en_design",
    "en-ted_en_business",
    "en-ted_en_global_issues",
    "en-wl_solar_power",
    "en-ted_en_playlist-what-s-really-floating-out-in",
    "en-ted_en_playlist-why-yes-actually-it-is-rocket",
    "en-ted_en_playlist-why-we-do-the-things-we-do",
    "en-khan-academy-videos_en_national-constitution-center-partner-videos"
  ]
}

**Comment by holta** [comment]
> can you paste in the contents of /library/www/html/home/menu.json

Or, @bernardroth can you please run [iiab-diagnostics](https://github.com/iiab/iiab/blob/master/scripts/iiab-diagnostics.README.md) and share the resulting sprunge.us (pastebin) URL?

**Reply by bernardroth** [reply]
http://sprunge.us/8LE5O3

**Comment by holta** [comment]
FYI @bernardroth Tim Moody is investigating, but please do send more details if you can, Thanks!

As background, an IIAB that was installed earlier this month has a `/library/www/html/common/assets/oer2go_catalog.json` which includes these Lines 1172-1194:

```
    "en-TED": {
      "age_range": "young adult,adult",
      "category": "educational,entertainment,justice",
      "description": "A collection of nearly 2000 short videos on a wide variety of thought-provoking TED (Technology, Entertainment, Design) talks.",
      "file_count": 1932,
      "index_mod_sample_url": null,
      "is_hidden": "No",
      "ksize": 39179028,
      "lang": "en",
      "logo_url": null,
      "moddir": "en-TED",
      "module_id": 17,
      "prereq_id": null,
      "prereq_note": "",
      "rating": "3.1",
      "rsync_url": "rsync://dev.worldpossible.org/rachelmods/en-TED",
      "source_url": "https://www.ted.com/",
      "title": "TED Talks",
      "type": "html",
      "version": "v0.2",
      "zip_ftp_url": "ftp://dev.worldpossible.org/zipped-modules/en-TED.zip",
      "zip_http_url": "http://dev.worldpossible.org/zipped-modules/en-TED.zip"
    },
```

Tangentially related:

- PR iiab/iiab-admin-console#466
- PR iiab/iiab-admin-console#467
- PR iiab/iiab-admin-console#468
- #2969

**Comment by tim-moody** [comment]
Here is what I think is the resolution. The menu definition file en-TED.json does not exist even though it was added to the home menu. I don't understand how this happened, but I have modified the option in Admin Console Content Menus -> Update Home Menu so that it will remove any menu item that does not exist.

@bernardroth you have actually surfaced a number of bugs here. In addition to this problem the en-TED module should not have been able to be downloaded as we prefer the zim version of TED talks. The ability to do so has also been removed. I hope this is not a problem as you were unable to download it anyway.

The obtain these fixes you will need to refresh the iiab-admin-console menu and run the install again.

**Reply by bernardroth** [reply]
How do I do this part "refresh the iiab-admin-console menu"?

**Reply by holta** [reply]
@bernardroth your IIAB install is more than 3 months old, so upgrading IIAB Admin Console might potentially be risky.

However, if you're willing to proceed (after backing up any critical items, and proceeding at your own risk) here's the general routine you'd run:

```
sudo su -    # Change to root
cd /opt/iiab/iiab-admin-console
git pull     # Make sure your IIAB is online for this step!
./install    # Takes up to (about) 5 minutes
```

**Reply by bernardroth** [reply]
I ran these but it didn't remove the missing item.

cd /opt/iiab/iiab-admin-console
sudo git pull
./install
cd /opt/iiab/iiab
sudo ./iiab-install --reinstall

Admin Console > Content Menus > Clicked "Update Home Menu"

**Reply by bernardroth** [reply]
I guess I could just edit this one /library/www/html/home/menu.json

**Reply by tim-moody** [reply]
that is the simplest if you don't want en-TED. 

Otherwise you could create /library/www/html/js-menu/menu-files/menu-defs/en-TED.json with the contents that @holta provided above.

**Reply by bernardroth** [reply]
Thanks for all the help. I created this file to fix it.

/library/www/html/js-menu/menu-files/menu-defs/en-TED.json
{
  "name": "en-TED",
  "lang": "en",
  "intended_use": "html",
  "moddir": "en-TED",
  "start_url": "",
  "title": "TED Talks",
  "logo_url": "ted.png",
  "description": "A collection of nearly 2000 short videos on a wide variety of thought-provoking TED (Technology, Entertainment, Design) talks.",
  "extra_description": "",
  "footnote": "",
  "edit_status": "generated",
  "change_ref": "generated",
  "change_date": "2021-09-19",
  "upload_flag": false,
  "download_flag": true
}

**Reply by holta** [reply]
> Thanks for all the help.

_Profound thanks @bernardroth — for your help explaining — and thereby helping everyone going forward._

**Comment by tim-moody** [comment]
please note that the menu property Check to Have the Menu Automatically Updated. must be checked for any changes to be made

**Comment by tim-moody** [comment]
still trying to understand exactly what happened.

There should have been two items in Admin Console Utilities -> Display Job Status that referenced en-TED

One something like usr/bin/rsync -Pavz --size-only rsync://dev.worldpossible.org/rachelmods/en-TED /library/working/rachel/
The other scripts/oer2go_install_step2.py en-TED

I'm expecting to see the first as FAILED
and the second as CANCELLED

**Reply by bernardroth** [reply]
Unfortunately the job listing is empty from the time I installed it last year. I reinstalled IIAB over it a few times since then.
